src/components/main-view/main-view.jsx

```diff
@@ -56,53 +53,67 @@ export class MainView extends React.Component {
       headers: { Authorization: `Bearer ${token}` }
     })
     .then(response => {
-      //assign the result to the state
-      this.setState({
-        movies: response.data
-      });
+      //#1
+      this.props.setMovies(response.data);
     })
     .catch(function (error) {
       console.log(error);
     });
   }
 
+  onLoggedOut() {
+    localStorage.removeItem("token");
+    localStorage.removeItem("user");
+    this.props.setUser('');
+    location.reload()

```

Comment:
There should be a better way to rerender the app once the user logs out. We can talk about it a bit at our next meeting. For now this is OK.